### PR TITLE
Added Swedish implementation

### PR DESCRIPTION
**Pull request by andagr**

### CODE REVIEW

**Comment by ploeh** [comment]
Thank you for your interest in contributing to Numsense. This pull request looks promising :+1:

When I run the build script, however, I get the following test failure:

```
    Ploeh.Numsense.SwedishExamples.tryOfEnglish returns correct result(swedish: "sexhundraarton", expected: 618) [FAIL]
      Swensen.Unquote.AssertionFailedException : Test failed:

      Some 618 = None
      false

      Stack Trace:
        Numsense.UnitTests\SwedishExamples.fs(125,0): at Ploeh.Numsense.SwedishExamples.tryOfEnglish returns correct result(String swedish, Int32 expected)
```

**Comment by andagr** [comment]
That's strange, all unit tests pass when I run them, I even tried with a fresh clone of my fork just to make sure. Do you have any suggestion on what I could try to replicate the error?

(I fixed the incorrectly named unit test btw, sorry about that.)

**Comment by ploeh** [comment]
Oh, I've just realised what went wrong. The string "sexhundraarton" contains a double-a, and in Danish, that's interpreted as "å" (even thought that actually hasn't been in use since 1948, except for person names).

My machine runs on `da-DK` locale, which is what's causing the error.

This could, in fact, be a general problem in this code base, only it hasn't materialised before now. We're going to have to figure out how to resolve this. Since this isn't your fault at all, I don't expect you to resolve it, but, on the other hand, if you know of a quick fix, then please share :smile:

**Comment by ploeh** [comment]
I have a solution; will share later. All tests now pass on my machine.

**Comment by andagr** [comment]
Well, nothing faster than 25 minutes ;), looking forward to see what you did.

**Comment by ploeh** [comment]
From a technical perspective I think we're good to go now :+1:

Although I do read Swedish, I'm not a native speaker, so I'd like someone who is more familiar with it to review that part, if at all possible. I don't expect any errors, but it's always good with a second pair of eyes :smile:

**Comment by ploeh** [comment]
I solicited a review on Twitter: https://twitter.com/ploeh/status/692089494556819456

Please retweet and spread the word :smile:

**Comment by kerams** [comment]
I'm not a native speaker either, so correct me if I'm wrong here. I know it's probably borderline incorrect, but you don't REALLY have to use `en`/`ett` with `hundra`, `tusen`, etc., do you? Supposing that it's true and that you had `tjugotvå tusen _ hundra fem` on the input for instance, your parser would incorrectly interpret the number as `22005` because of [this line](https://github.com/andagr/Numsense/blob/master/Numsense/Swedish.fs#L98).

**Comment by andagr** [comment]
It's perfectly fine to omit "ett" when saying e.g. 105, but when it's a higher number such as 22105 it sounds very odd. But with that said I'm not exactly sure if there is a grammatical rule that specifically disallows it, and 22005 is definately wrong (None would be less bad).

So, I added a fix for this case, good catch!

**Comment by andagr** [comment]
@ploeh 
I added test cases for how years are naturally pronounced, such as "1978" which can be said as nittion-hundra-sjuttio-åtta (nineteen-hundred-seventy-eight in english), since that just worked without any changes to the code. Is this something you had in mind of supporting with this lib or should I remove the test cases again?

**Comment by kerams** [comment]
@andagr, no problem. Both `tusen` and `miljon` suffer from the same issue, however. I too had to deal with this in my PR :).

**Comment by andagr** [comment]
Thanks for the input, I think I've fixed all the cases you mentioned now.

**Comment by kerams** [comment]
:+1:

**Comment by ploeh** [comment]
> I added test cases for how years are naturally pronounced, such as "1978" which can be said as nittion-hundra-sjuttio-åtta (nineteen-hundred-seventy-eight in english), since that just worked without any changes to the code. Is this something you had in mind of supporting with this lib or should I remove the test cases again?

I can see that you've added them to the test cases for `tryParseSwedish`, which only demonstrates that it's a [Tolerant Reader](http://martinfowler.com/bliki/TolerantReader.html). In most cases, I apply [Postel's Law](https://en.wikipedia.org/wiki/Robustness_principle) to API design, and that's also the case here.

**tl;dr:** :+1:

**Comment by ploeh** [comment]
It doesn't seem as though there are any objections from Swedish-speaking people, so I'm ready to move on with this pull request.

As you may have noticed, the branch is no longer automatically mergeable. That's not your fault, but it's best when the author of a pull request gets a chance to address merge conflicts. Could you please do that?

Post here when done, so that I get a notification, and I'll pull :smile:

**Comment by andagr** [comment]
Sure thing, and done! :)

**Comment by ploeh** [comment]
Thank you for your contribution! It's now live as [Numsense 0.7.0](https://www.nuget.org/packages/Numsense/0.7.0).

**Comment by andagr** [comment]
Thanks!
